src/robothub/replay_camera.py

```diff
@@ -0,0 +1,542 @@
+        if path.is_file():
+            self._path_type = PathType.VIDEO
+            return path
+        elif path.is_dir():
+            self._path_type = PathType.IMAGE_DIRECTORY
+            return path
+        else:
+            logging.error(f"provided file: {src} is not a file or not a dir")
+
+    def _create_cam_input(self, pipeline: dai.Pipeline, stream_name: str):
+        node = pipeline.createXLinkOut()
+        node.setStreamName(stream_name)
+        self._nodes[stream_name] = node
+        return node.input
+
+    def _create_cam_output(self, pipeline: dai.Pipeline, stream_name: str, max_data_size: int | None = None):
+        node = pipeline.create(dai.node.XLinkIn)
+        node.setStreamName(stream_name)
+        if max_data_size is not None:
+            node.setMaxDataSize(max_data_size)
+        self._nodes[stream_name] = node
+        return node.out
+
+    def send_video_frames(self, device: dai.Device):
+        self._input_control_queue = device.getOutputQueue(name=self.INPUT_CONTROL_STREAM_NAME)
+        self._input_config_queue = device.getOutputQueue(name=self.INPUT_CONFIG_STREAM_NAME)
+
+        self._raw_queue = device.getInputQueue(name=self.RAW_STREAM_NAME)
+        self._isp_queue = device.getInputQueue(name=self.ISP_STREAM_NAME)
+        self._video_queue = device.getInputQueue(name=self.VIDEO_STREAM_NAME)
+        self._still_queue = device.getInputQueue(name=self.STILL_STREAM_NAME)
+        self._preview_queue = device.getInputQueue(name=self.PREVIEW_STREAM_NAME)
+
+        if self._path_type == PathType.VIDEO:
+            self._cap = cv2.VideoCapture(str(self._path))
+        else:
+            self._cap = ImageDirectoryCapture(self._path)  # type: ignore
+
+        sequence_number = 0
+
+        send_video_frames_start = time.time()
+
+        def create_img_frame(
+            data: np.ndarray,
+            width: int,
+            height: int,
+            type: dai.RawImgFrame.Type,
+            sequence_number: int,
+            timestamp: timedelta,
+        ):
+            img_frame = dai.ImgFrame()
+            img_frame.setType(type)
+            img_frame.setData(data.flatten())
+            img_frame.setTimestamp(timestamp)
+            img_frame.setSequenceNum(sequence_number)
+            img_frame.setWidth(width)
+            img_frame.setHeight(height)
+            return img_frame
+
+        def get_next_frame():
+            if not self._cap.isOpened():
+                return None
+
+            frame = None
+            for _ in range(2):
+                ok, frame = self._cap.read()
+                if not ok and self._loop:
+                    self._cap.release()
+                    if self._path_type == PathType.VIDEO:
+                        self._cap = cv2.VideoCapture(str(self._path))
+                    else:
+                        self._cap.reset()  # type:ignore
+                    continue
+                break
+
+            return frame
+
+        while True:
```

Comment:
this ties to my worry that its not possible to terminate this loop unless it stops on its own.
I think we should use here
```
while rh.app_is_running:
    pass
```
if we want to give the user the option to turn it off manually, then add a second flag